pkg/step/feature/derivation.go

```diff
@@ -338,11 +338,37 @@ func deriveFeatureColumn(fcMap ColumnMap, columnTargets []string, fmMap FieldDes
 			fcMap[target] = make(map[string][]ir.FeatureColumn)
 			fcTargetMap = fcMap[target]
 		}
+		fcMap[target] = make(map[string][]ir.FeatureColumn)
 		for f := range fcTargetMap {
 			if _, ok := selectFieldTypeMap[f]; !ok {
-				return fmt.Errorf("Unknown column '%s' in 'column clause'", f)
+				if len(fcTargetMap[f]) != 1 {
+					return fmt.Errorf("cannot expand '%s' in 'column clause'", f)
+				}
+				// Try as regex to match the selected fields
+				r, e := regexp.Compile("(?i)^" + f + "$")
+				if e != nil {
+					return fmt.Errorf("unknown column '%s' in 'column clause'", f)
+				}
+				hasMatch := false
+				for sf := range selectFieldTypeMap {
+					if r.MatchString(sf) {
+						applied, err := fcTargetMap[f][0].ApplyTo(fdMap[sf])
+						if err != nil {
+							return err
+						}
+						fcMap[target][sf] = []ir.FeatureColumn{applied}
+						hasMatch = true
+					}
+				}
+				if !hasMatch {
+					return fmt.Errorf("'%s' in 'column clause' does not match any selected fields", f)
+				}
+				delete(fdMap, f)
```

Comment:
In the `if` statement that starts at L343 and ends at L367,  we assume that `f` is a regular expression because it is not in the selected fields `selectFieldTypeMap`.
After the `if !hasMatch` condition in L363-365, we are sure `f` is a regex that matches some fields in `selectFieldTypeMap`, as a result, the original `f` entry is deleted from `fdMap` because it's not a real field.

Reply:
I see.